src/Common/BitonicSort.h

```diff
@@ -0,0 +1,323 @@
+#pragma once
+
+#include <string.h>
+#if !defined(__APPLE__) && !defined(__FreeBSD__)
+#include <malloc.h>
+#endif
+
+#ifdef __APPLE__
+#include <OpenCL/opencl.h>
+#else
+#include <CL/cl.h>
+#endif
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <cstdint>
+#include <map>
+#include <type_traits>
+
+#include <ext/bit_cast.h>
+#include <Core/Types.h>
+#include <Core/Defines.h>
+
+#include "OpenCLBasics.cpp"
+
+#define CL_USE_DEPRECATED_OPENCL_1_2_APIS
```

Comment:
It does not make sence if definaed after `#include <CL/cl.h>`. We do not need it by default and should use OpenCL version from cmake to detect if it's old.

Reply:
Fixed with checking `CL_VERSION_2_0`